java/src/main/java/ai/onnxruntime/OnnxRuntime.java

```diff
@@ -115,15 +113,10 @@ private static void cleanUp(File file) {
   /**
    * Check if we're running on Android.
    *
-   * @return True if the {@code android.app.Activity} class can be loaded, false otherwise.
+   * @return True if the property java.vendor equals The Android Project, false otherwise.
    */
   static boolean isAndroid() {
-    try {
-      Class.forName("android.app.Activity");
-      return true;
-    } catch (ClassNotFoundException e) {
-      return false;
-    }
+    return System.getProperty("java.vendor", "generic").equals("The Android Project");
```

Comment:
Does this cover all versions of Android and android based OSs? It seems like trying to load something from the Android class library is more reliable, so in what circumstances does it fail?

Reply:
According this document, https://developer.android.com/reference/java/lang/System#getProperties()
This is added since Android API level 1 which is the beginning of Android,
Currently load class library is not failing however get system properties is a preferred way to tell if we are running Android

Reply:
Ok. Those are example properties though, not guaranteed by all android instances, at least according to those docs.

Reply:
Both load class and getProperty are not perfect, although I personally prefer getProperty (which has not changed in the past 10 years) than load system class (which may also be changed by different vendors), but I don't really have strong opinion on this. I guess both ways will work.

Reply:
Fair enough. I guess if LineageOS, FireOS or the other Android forks changed those things it would probably break a bunch of apps if other people detect it this way, so it's probably the same.